### ISSUE TITLE
Explore how to show garage visits in the "Vehicle activity" dashboard

### ISSUE THREAD

**Issue by sobuos** [OS2fleetoptimiser/OS2fleetoptimiser#36]
Some users have requested that the "vehicle activity" dashboard show when vehicles are at a garage for repairs. Currently, vehicles only appear on the dashboard when they are completely stationary. However, when they are moved between garages, they are displayed as driving a very short distance.

**Comment by sobuos** [comment]
This issue was prioritized on the development meeting in the user group on February 19. Has been postponed up until this point because it was awaiting a scheduled maintenance of the aggregation module.

An important note is that vehicles must be able to have several pre-defined garages. 

@andreasDroid when are you and your team planning on initiating this development issue?

**Comment by andreasDroid** [comment]
Hi @sobuos ,

Just to be clear on the note that vehicles must be able to have several pre-defined garages.
If I understand it correctly, the users' want to be able to maintain a list of garages to each vehicle? 

If this is the case, it would be quite a big extension to the proposed suggestion which included: 
- Add new workshops/garages or edit them from the Locations page
- User defined definition of visit: e.g. 8 hours or 4 hours. 
- Aggregation will look at registered trips and mark when a vehicle is within the radius of a registered workshop for the user defined amount of time 
- Driving activity will display the days with workshop visits 

Let me know if I understand it wrongly. Since I'll argue that maintaining a list of garages for each vehicles is not only extensive development, but also tedious to maintain for the users.

[værkstedsbesøg1.pdf](https://github.com/user-attachments/files/27308639/vaerkstedsbesog1.pdf)


Could you also help reaching out to Fleetmanagers to help us get example of workshop visits. License plates, workshop address and workshop visit date would be sufficient :)

**Comment by andreasDroid** [comment]
Elaborating the details for @o-droids and the team. 

### Frontend
**New page**
- New page under configuration named "Værksteder".
- Includes a list of registered workshops. 
- Allows adding of new workshops.
- Allows editing and deletion of registered workshops. 
- Allows defining a global setting "Minimum tid for registrering af besøg" - @sobuos please help define a good naming for this setting
- Explanation of what the workshop functionality is and its purpose
- Optional: functionality to list all workshop visits in a given periode - @sobuos perhaps this is a nice addition, which would not require a lot of extra implementation. Will you confirm if we should also implement this? 

**Driving activity dashboard**
- Add legends displaying what the colors denote: white (activity), red (no activity), grey (active roundtrip, no driving), new color (workshop visit)
- Add functionality in graph to display the workshop visits
- Implement API integration to backend that pulls workshop visits for selected filters

### Datamodel
- addition of a workshop table that holds: id, latitude, longitude, name/address
- addition of a workshop visit table that holds: reference to car_id, reference to workshop id, start time of visit, end time of visit, duration of visit
- NOTE: just add the duration of the visit, allows for dynamic changing of threshold

### Aggregation
- Functionality to look at complete gps log lists to check workshop visits. Be aware: it is not enough to use the delta load for this. We need to have a wider picture of the driving in order to confidently classify the workshop visits. Suggestion to load latest week of driving and classify on basis of that. Check existing workshop visits as this approach will see the same visit multiple times. Checks should happen before any aggregation and have it's own pipeline. 

### Backend
- Get endpoint to list workshops
- Post endpoint to create workshops
- Patch endpoint to edit workshops
- Delete endpoint to delete workshops
- Get endpoint to list workshop visits (incl. filters)
